File: 0036-valid-sudoku/0036-valid-sudoku.cpp
class Solution {
public:
    bool isValidSudoku(vector<vector<char>>& board) {
        bool rows[9][9] = {false};
        bool cols[9][9] = {false};
        bool boxes[9][9] = {false};

        for (int y = 0; y < 9; y++) {
            for (int x = 0; x < 9; x++) {
                if (board[y][x] == '.') {
                    continue;
                }

                int num = board[y][x] - '0' - 1;
                int box = y/3 * 3 + x/3;

                if (rows[y][num] || cols[x][num] || boxes[box][num]) {
                    return false;
                }

                rows[y][num] = true;
                cols[x][num] = true;
                boxes[box][num] = true;
            }
        }

        return true;
    }
};

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        for(int i = 0; i < 9; i++){
            Set<Character> set = new HashSet<>();
            for(int j = 0; j < 9; j++){
                char item = board[i][j];
                if(item != '.' && !set.add(item)){
                    return false;
                }
            }
        }
        for(int i = 0; i < 9; i++){
            Set<Character> set = new HashSet<>();
            for(int j = 0; j < 9; j++){
                char item = board[j][i];
                if(item != '.' && !set.add(item)){
                    return false;
                }
            }
        }
        int[][] valids = {{0,0},{0,3},{0,6},
        {3,0},{3,3},{3,6},
        {6,0},{6,3},{6,6}};
        for(int[] valid : valids){
            Set<Character> set = new HashSet<>();
            for(int i = valid[0]; i < valid[0] + 3; i++){
                for(int j = valid[1]; j < valid[1] + 3; j++){
                    char item = board[i][j];
                    if(item != '.' && !set.add(item)){
                        return false;
                    }
                }
            }
        }
        return true;
    }
}